src/ostree.cc

```diff
@@ -55,6 +55,29 @@ data::InstallOutcome OstreeManager::pull(const Config &config, const KeyManager
   return data::InstallOutcome(data::OK, "Pulling ostree image was successful");
 }
 
+data::InstallOutcome OstreeManager::find_or_pull(const Config &config, const KeyManager &keys, const std::string &refhash) {
+  OstreeRepo *repo = NULL;
+  const char *const commit_ids[] = {refhash.c_str()};
+  GCancellable *cancellable = NULL;
+  GError *error = NULL;
+  GVariantBuilder builder;
+  GVariant *options;
```

Comment:
`commit_ids`, `builder`, and `options` are not actually used here. But does it make sense to merge this into the `pull` function anyway? Normally I wouldn't advocate for making functions longer, but there are some duplicated calls and it's the only place `pull` is used. Up to you.